python-package/xgboost/core.py

```diff
@@ -237,18 +254,18 @@ def _cuda_array_interface(data) -> bytes:
     return interface_str
 
 
-def ctypes2numpy(cptr, length, dtype) -> np.ndarray:
+def ctypes2numpy(cptr: ctypes.pointer, length: int, dtype: Type[np.number]) -> np.ndarray:
     """Convert a ctypes pointer array to a numpy array."""
-    ctype = _numpy2ctypes_type(dtype)
-    if not isinstance(cptr, ctypes.POINTER(ctype)):
+    ctype: Type[supported_ctypes] = _numpy2ctypes_type(dtype)
+    if not isinstance(cptr, ctypes.POINTER(ctype)):  # should we use `issubclass`?
         raise RuntimeError(f"expected {ctype} pointer")
     res = np.zeros(length, dtype=dtype)
-    if not ctypes.memmove(res.ctypes.data, cptr, length * res.strides[0]):
+    if not ctypes.memmove(res.ctypes.data, cptr, length * res.strides[0]):  # "memmove" does not return a value?
```

Comment:
I think that's probably a bug in mypy.

Reply:
You were right. By the way, the maintainers of typeshed were able to fix this bug in a flash.